src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3739,6 +3749,34 @@ static NullableFlowState getResultState(BoundCollectionExpression node, Collecti
             }
         }
 
+        public override BoundNode? VisitCollectionExpressionSpreadElement(BoundCollectionExpressionSpreadElement node)
+        {
+            VisitRvalue(node.Expression);
+            var result = _visitResult;
+
+            if (node.Conversion is BoundConversion { Conversion: var conversion })
+            {
+                Debug.Assert(node.ExpressionPlaceholder is { });
+                AddPlaceholderReplacement(node.ExpressionPlaceholder, node.Expression, result);
+                VisitForEachExpression(
+                    node,
+                    node.Conversion,
+                    conversion,
+                    node.ExpressionPlaceholder,
+                    node.EnumeratorInfoOpt,
+                    awaitOpt: null);
```

Comment:
It seems like we aren't looking at the `node.IteratorBody`? Perhaps there isn't a need to do that if things look right in the `VisitForEachExpression` itself?

Reply:
Good catch, we should be visiting `node.IteratorBody`. Added to the caller.